diff --git a/node_modules/@expo/browser-polyfill/src/DOM/HTMLImageElement.js b/node_modules/@expo/browser-polyfill/src/DOM/HTMLImageElement.js
index 235f341..6d09302 100644
--- a/node_modules/@expo/browser-polyfill/src/DOM/HTMLImageElement.js
+++ b/node_modules/@expo/browser-polyfill/src/DOM/HTMLImageElement.js
@@ -1,5 +1,5 @@
 import { Image } from 'react-native';
-import { FileSystem } from 'expo';
+import * as FileSystem from 'expo-file-system';
 const { writeAsStringAsync, documentDirectory } = FileSystem;
 const EncodingType = FileSystem.EncodingType || FileSystem.EncodingTypes;
 
