diff --git a/node_modules/react-native/React/Modules/RCTAsyncLocalStorage.m b/node_modules/react-native/React/Modules/RCTAsyncLocalStorage.m
index 508ec11..a464352 100644
--- a/node_modules/react-native/React/Modules/RCTAsyncLocalStorage.m
+++ b/node_modules/react-native/React/Modules/RCTAsyncLocalStorage.m
@@ -200,7 +200,8 @@ - (NSString *)_filePathForKey:(NSString *)key
 
 - (NSDictionary *)_ensureSetup
 {
-  RCTAssertThread(RCTGetMethodQueue(), @"Must be executed on storage thread");
+  // Don't assert this. We use this at startup, not on the storage thread, to read AsyncStorage before init'ing Expo and React Native
+  // RCTAssertThread(RCTGetMethodQueue(), @"Must be executed on storage thread");
 
 #if TARGET_OS_TV
   RCTLogWarn(@"Persistent storage is not supported on tvOS, your data may be removed at any point.");
